frontend/src/components/Gym/ConfirmationModal.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import Button from '../Button'
+import styled from 'styled-components'
+
+import 'antd/dist/antd.css'
+
+const OverlayContainer = styled.div<{ overlayBackground: string; isModalOpen: boolean }>`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  min-height: 100vh;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: ${(props) => (props.overlayBackground ? props.overlayBackground : 'rgba(0, 0, 0, 0.3)')};
+  z-index: 999;
+  ${(props) => !props.isModalOpen && `display: none;`}
+`
+
+const MainContainer = styled.div<{ isModalOpen: boolean }>`
+  position: fixed;
+  background-color: #fff;
+  width: 90vw;
+  border-radius: 15px;
+  margin-left: auto;
+  margin-right: auto;
+  padding: 15px;
+  box-shadow: 0 2px 5px 1px #888888;
+  z-index: 1000;
+  display: flex;
+  flex-direction: column;
+  ${(props) => !props.isModalOpen && `display: none;`}
+`
+
+const TitleContainer = styled.div<{ flex?: boolean }>`
+  ${(props) => props.flex && 'display: flex;'}
+  text-align: center;
+`
+
+const TitleText = styled.h1`
+  font-family: Lato;
+  font-size: 16px;
+  font-weight: 400;
+`
+
+const DescriptionText = styled.p`
+  font-weight: 250;
+  font-size: 14px;
+  padding-top: 0.5rem;
+  color: #33363a;
+  text-align: justify;
+`
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-evenly;
+  width: 100%;
+  margin-top: 10px;
+`
+
+type Props = {
+  title: string
+  description?: string | JSX.Element
+  // Left Button Props
+  leftButtonType: 'primary' | 'secondary'
+  leftButtonText: string
+  onLeftButtonClick: React.MouseEventHandler<HTMLButtonElement>
+  // Right Button Props
+  rightButtonType: 'primary' | 'secondary'
+  rightButtonText: string
+  onRightButtonClick: React.MouseEventHandler<HTMLButtonElement>
+  // Overlay Props
+  // Pass function to do something when the overlay around the modal is clicked
+  onOverlayClick?: React.MouseEventHandler<HTMLDivElement>
+  // Pass a style for custom overlay
+  overlayBackground?: string
+  // Custom styles
+  top?: number | string
+  bottom?: number | string
+  right?: number
+  left?: number
+  flex?: boolean
+  style?: React.CSSProperties
+  rightButtonHtmlType?: 'button' | 'submit' | 'reset' | undefined
+  // Pass this into Modal to render/unrender the modal
+  isModalOpen: boolean
+  // Use this to toggle overlay
+  setModalOpen: (boolean) => void
+}
+
+export function ConfirmationModal(props: Props) {
+  // Checks for custom buttom styles, otherwise uses default
+  const defaultOverlayBackground = props.overlayBackground ?? 'rgba(0, 0, 0, 0.3)'
+
+  return (
+    <>
+      <OverlayContainer
+        onClick={props.onOverlayClick}
+        overlayBackground={defaultOverlayBackground}
+        isModalOpen={props.isModalOpen ?? true}
+      />
+      <MainContainer
+        isModalOpen={props.isModalOpen ?? true}
```

Comment:
consider returning `<></>` if `props.isModalOpen === false` instead? so there is no need to pass isModalOpen into the OverlayContainer and MainContainer components?

maybe something like `{props.isModalOpen ? <><OverlayContainer.... : <></>}` ?

Reply:
also since isModalOpen is a boolean it can only be true or false, is it necessary for the `??` operator?